### ISSUE TITLE
Ram explodes when training a very simple model with accelerate tpu

### ISSUE THREAD

**Issue by dercodeKoenig** [huggingface/accelerate#4010]
i try to get a very simple model to run on tpu, the model is just 2 rrdb blocks currently, can train fast on cpu, but when running on tpu it takes about 40gb ram before the loop starts, 80gb when the loss prints, and then crashes using 330+gb ram before the second step starts. using kaggle tpuv5e-8
```Python

class FlowModel:
    def __init__(self, model, optimizer, accelerator: Accelerator):
        self.model = model
        self.optimizer = optimizer
        self.accelerator = accelerator
        self.device = accelerator.device
        self.cri_pix = torch.nn.MSELoss()

    def optimize_parameters(self, gt, lq):
        # Use the accumulate context manager to handle sync and stepping
        with self.accelerator.accumulate(self.model):
            with torch.no_grad():        
                
                target_gt = (gt * 2.0) - 1.0
                lr_input = (lq * 2.0) - 1.0

                b, c, h, w = target_gt.shape

                noise = torch.randn_like(target_gt)
                t = torch.rand((b, 1, 1, 1), device=self.device)

                # 2. Apply Flow logic
                xt = t * target_gt + (1.0 - t) * noise
                drift = torch.randn_like(xt) * 0.02 * (1.0 - t)
                xt = xt + drift

                v_target = (target_gt - xt) / (1.0 - t).clamp(min=1e-3)

                model_input = torch.cat([xt, lr_input], dim=1)
                time_input = t.view(b)

            # 3. Forward Pass with Accelerator's autocast
            with self.accelerator.autocast():
                v_pred = self.model(model_input, time_input)
                l_total = self.cri_pix(v_pred, v_target)

            # 4. Backward Pass
            # Accelerator handles the loss scaling for accumulation automatically
            self.accelerator.backward(l_total)

            self.optimizer.step()
            self.optimizer.zero_grad()
            
            return l_total.detach().item()

def main():
    accelerator = Accelerator(gradient_accumulation_steps=1)
    
    dataset = RealESRGANDataset(opt["Dataset"])
    
    train_loader = DataLoader(
        dataset=dataset,
        num_workers=0, 
        batch_size=1,        # How many images per batch
        shuffle=True,        # Randomize order every epoch      
        #pin_memory=True      # Speeds up transfer from CPU to GPU
    )
    
    model = PixelFlowV1(**opt["Model"])
    print(accelerator.device, "model created")
    
    optimizer = torch.optim.Adam(model.parameters(), lr=1e-4)
    print(accelerator.device, "optimizer created")
    
    # Prepare model, optimizer, dataloader as usual
    model, optimizer, train_loader = accelerator.prepare(
        model, optimizer, train_loader
    )
    
    srflow = FlowModel(model, optimizer, accelerator)

    print(accelerator.device, "starting training")
    for batch in train_loader:
        gt, lq = batch["gt"], batch["lq"]
        print(accelerator.device, gt.shape, lq.shape)
        loss = srflow.optimize_parameters(gt,lq)
        print("loss", loss)
        #print("step complete")
        #import torch_xla.debug.metrics as met
        #print(met.metrics_report())
        #break

    #accelerator.save_state("state")
    accelerator.end_training()
``` 

``` 
import os
try:
    os.environ.pop('TPU_PROCESS_ADDRESSES')
except:
    pass


!accelerate launch --tpu --mixed_precision "bf16" --dynamo_backend no --num_processes 0 --main_training_function main train.py
``` 

before crashing from out of ram, it prints this as expected
`
xla:0 torch.Size([1, 3, 256, 256]) torch.Size([1, 3, 256, 256])
xla:0 torch.Size([1, 3, 256, 256]) torch.Size([1, 3, 256, 256])
xla:0 torch.Size([1, 3, 256, 256]) torch.Size([1, 3, 256, 256])
xla:0 torch.Size([1, 3, 256, 256]) torch.Size([1, 3, 256, 256])
xla:0 torch.Size([1, 3, 256, 256]) torch.Size([1, 3, 256, 256])
xla:0 torch.Size([1, 3, 256, 256]) torch.Size([1, 3, 256, 256])
xla:0 torch.Size([1, 3, 256, 256]) torch.Size([1, 3, 256, 256])
xla:0 torch.Size([1, 3, 256, 256]) torch.Size([1, 3, 256, 256])
loss 1.2578125
loss 1.1875
loss 1.34375
loss 1.25
loss 1.09375
loss 1.25
loss 1.1953125
loss 1.421875
`

**Comment by dercodeKoenig** [comment]
when i make the model even smaller, i mean extremly extremly small, it runs with 240gb ram consumed

**Comment by dercodeKoenig** [comment]
the model / code appears to run normal when running on 2 kaggle gpus

**Comment by dercodeKoenig** [comment]
https://www.kaggle.com/code/bpwqsdd/notebookc992b7edc8

**Comment by dercodeKoenig** [comment]
i think this is nothing to do with accelerate
